### DISCUSSION TITLE
What is the right way to obtain SOC values during the experiment?

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2936] [Q&A] [answered]
I can see SOC being used in the source code as a variable, as well as initial SOC, but how can I change initial SOC and extract SOC from solution within a script?

**Comment by brosaplanella** [comment, marked as answer]
There is no SoC variable built in PyBaMM as there are multiple definitions, but you can get the SoC according to your own definition by using postprocessing the solution (e.g. dividing the `Discharge capacity [A.h]` by the total capacity).

**Reply by cyasing** [reply]
You mean `(Discharge capacity from solution) / (Nominal cell capacity from parameters)` right?

**Reply by brosaplanella** [reply]
Yes, though if the battery age you might want to update the value for the cell capacity.

**Reply by cyasing** [reply]
By step solving or summary variables, then?

**Reply by brosaplanella** [reply]
Whichever way you choose to define your capacity (different people use different definitions).

**Reply by cyasing** [reply]
Right
